fbsat-core/src/main/kotlin/ru/ifmo/fbsat/core/scenario/negative/NegativeScenarioTree.kt

```diff
@@ -86,6 +87,7 @@ class NegativeScenarioTree(
     inner class Node(
         override val element: ScenarioElement,
         override val parent: Node?,
+        var label: Int = 0,
```

Comment:
I am against such structural changes. In my experience, it will become unmaintainable pretty quickly. I would advice using external data structures, such as `Map<Node, Int>` for these kinda tasks.
We also could adhoc either `val content: Any` field for `Node`, or maybe somehow reuse `Solver::context` (though, currently Tree is completely disconnected from Solver), but let's be simple: if you want to supplement the core data structure (Tree, in this case) with additional information, then either wrap it externally however and with whatever you want, or make sure that the change plays perfectly with the existing codebase (which is slightly a mess nowadays, so let's not make it worse).